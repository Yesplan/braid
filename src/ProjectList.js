import React from 'react';
import { Link } from 'react-router-dom';

const colorNames = ['is-primary', 'is-link', 'is-info', 'is-danger'];

const projectColor = index => colorNames[index % colorNames.length];

const ProjectList = ({ projects }) => (
  <div className="section">
    <h1 className="title is-3">Select project</h1>

    <div
      style={{
        display: 'grid',
        gridTemplateColumns: '1fr 1fr 1fr',
        gridGap: '1.5rem'
      }}
    >
      {projects.map((project, index) => (
        <Link
          to={`/projects/${project.projectId}`}
          className={`notification ${projectColor(index)} is-marginless`}
          key={project.projectId}
        >
          <p className="title">{project.projectName}</p>
          <p className="subtitle">{project.role}</p>
        </Link>
      ))}
    </div>
  </div>
);

export default ProjectList;
