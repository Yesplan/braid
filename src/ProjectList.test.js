import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { render } from '@testing-library/react';
import { ProjectList } from './ProjectList';

describe('ProjectList', () => {
  const projects = [
    {
      projectName: 'Dominate the tri-state-area',
      projectId: 1,
      projectRole: 'Supreme leader'
    },
    {
      projectName: 'Next level grocery shopping',
      projectId: 2,
      projectRole: 'Bystander'
    },
    {
      projectName: 'Afternoon nap',
      projectId: 3,
      projectRole: 'Member'
    },
    {
      projectName: 'Side-project',
      projectId: 4,
      projectRole: 'Owner'
    },
    {
      projectName: 'Death star construction',
      projectId: 5,
      projectRole: 'Janitor'
    }
  ];

  it('renders single project according to snapshot', () => {
    const { container } = render(
      <Router>
        <ProjectList projects={[projects[0]]} />
      </Router>
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders alternating colors', () => {
    const { container } = render(
      <Router>
        <ProjectList projects={projects} />
      </Router>
    );

    expect(container.querySelector('a[href="/projects/1"]')).toHaveClass(
      'is-primary'
    );

    expect(container.querySelector('a[href="/projects/2"]')).toHaveClass(
      'is-link'
    );

    expect(container.querySelector('a[href="/projects/3"]')).toHaveClass(
      'is-info'
    );

    expect(container.querySelector('a[href="/projects/4"]')).toHaveClass(
      'is-danger'
    );

    expect(container.querySelector('a[href="/projects/5"]')).toHaveClass(
      'is-primary'
    );
  });
});
