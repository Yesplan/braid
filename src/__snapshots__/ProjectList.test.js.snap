// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectList renders single project according to snapshot 1`] = `
<div
  class="section"
>
  <h1
    class="title is-3"
  >
    Select project
  </h1>
  <div
    style="display: grid; grid-template-columns: 1fr 1fr 1fr;"
  >
    <a
      class="notification is-primary is-marginless"
      href="/projects/1"
    >
      <p
        class="title"
      >
        Dominate the tri-state-area
      </p>
      <p
        class="subtitle"
      />
    </a>
  </div>
</div>
`;
